### PR TITLE
Misc. changes from break

### PR DESCRIPTION
**Pull request by 3141592654**
@mhrmm The changes, which are quite miscellaneous, are more easily digested by looking at the individual commits. Over the past week I've run ~50 experiments. A selection of those are in Overleaf.

### CODE REVIEW

File: pkmn.py [hunk 1]
```diff
@@ -70,54 +66,69 @@ def policy_forward(x, cur_model):
...
     # The latter is not strictly necessary, but it will make our lives easier
     return pvec, h, h2
 
 def policy_backward(bookkeeper):
     # Stack all the data from the bookkeeper.
-    xs = np.vstack(bookkeeper.xs).T
-    hs = np.vstack(bookkeeper.hs).T
-    h2s = np.vstack(bookkeeper.h2s).T
-    pvecs = np.vstack(bookkeeper.pvecs).T
+    xs = bookkeeper.xs#np.vstack(bookkeeper.xs)#.T
```
Comment (mhrmm) [inline pkmn.py:75]: Are you planning to revert to this commented code frequently? Otherwise it's generally good form to eliminate it and just let Git remember it if you need it again.

**Review by mhrmm — COMMENTED** [review]
LGTM. You may want to clean out some of the commented code, if it's truly obsolete.

File: pkmn.py (hunk 1 above)
Comment (3141592654) [reply pkmn.py:75]: That particular comment I haven't decided what to do with yet. Back when we used a single neural net we could take all of the pvecs and all of the h2s and mash them up into vectors/matrices respectively, and then pass them to backprop_relu_hidden_layer along with our only set of weights. Now we have 36 sets of weights, so we can't do that. Right now I am going through each game individually and calculating the gradients for each game. The options going forward are:
1. Leave it as it is and delete the comment. It is slow, but we have other priorities
2. Split up the variables based on what model they used, then vstack as before
3. Re-implement whole thing in C++ for performance
I've added dealing with this to the todo list.
